### PR TITLE
Vault integration: add remaining repo CRUDL operations for credential store

### PR DESCRIPTION
**Pull request by mgaffney**
Adds the following methods to the repository:

* `LookupCredentialStore`
* `UpdateCredentialStore`
* `DeleteCredentialStore`
* `ListCredentialStores`

### CODE REVIEW

**Review by louisruch — APPROVED** [review]
LGTM with some minor comments!

File: internal/db/schema/migrations/postgres/5/02_vault_credential.up.sql [hunk 1]
```diff
@@ -262,4 +276,76 @@ begin;
...
+            cert.certificate        as client_cert,
+            cert.certificate_key    as ct_client_key, -- encrypted
+            cert.key_id             as client_key_id
+       from credential_vault_store store
+  left join current_tokens token
+         on store.public_id = token.store_id
+  left join credential_vault_client_certificate cert
+         on store.public_id = cert.store_id;
+  comment on view credential_vault_store_client_private is
+    'credential_vault_store_client_private is a view where each row contains a credential store and the credential store''s data needed to connect to Vault. '
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/db/schema/migrations/postgres/5/02_vault_credential.up.sql:324]: It is an escape character.

File: internal/credential/vault/vault.go [hunk 2]
```diff
@@ -122,3 +122,11 @@ func (c *client) LookupToken() (*vault.Secret, error) {
 	}
 	return t, nil
 }
+
+// SwapToken replaces the token in the Vault client with t and returns the
+// token that was replaced.
+func (c *client) SwapToken(t string) string {
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/credential/vault/vault.go:128]: I will update. ty!

File: internal/credential/vault/repository_credential_store.go [hunk 3]
```diff
@@ -215,3 +230,584 @@ func validateTokenLookup(op errors.Op, s *vault.Secret) error {
...
+		case strings.EqualFold("ClientCertificate", f):
+		default:
+			filteredNullFields = append(filteredNullFields, f)
+		}
+	}
+
+	oplogWrapper, err := r.kms.GetWrapper(ctx, cs.ScopeId, kms.KeyPurposeOplog)
+	if err != nil {
+		return nil, db.NoRowsAffected,
+			errors.Wrap(err, op, errors.WithCode(errors.Encrypt), errors.WithMsg("unable to get oplog wrapper"))
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/credential/vault/repository_credential_store.go:581]: ty!

File: internal/credential/vault/repository_credential_store_test.go [hunk 4]
```diff
@@ -208,3 +206,1148 @@ func TestRepository_CreateCredentialStoreNonResource(t *testing.T) {
...
+
+func TestRepository_lookupPrivateCredentialStore(t *testing.T) {
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrapper := db.TestWrapper(t)
+
+	tests := []struct {
+		name    string
+		tls     TestVaultTLS
+		wantErr errors.Code
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/credential/vault/repository_credential_store_test.go:303]: ty!

**Pull request merged**